apps/main/src/GithubApp/componets/Header/Header.tsx

```diff
@@ -18,11 +18,11 @@ export const Header: FC<HeaderPropertiesInterface> = memo(({ handleSetLicense, h
         className={classes.headerField}
         name="search"
         placeholder="Search by repository name"
-        onChange={handleSetRepositoryName}
+        {...form.getInputProps('repositoryName')}
```

Comment:
Some tests failed on this line

Reply:
Hey... Can you help me define the type of getInputProps() in Header.types.rs? I am not getting how to define its type...

Reply:
Yes of course

You have a correct types here
https://github.com/Malay-dev/nx-vite-react-ts-mantine-boilerplate/blob/mantine-hook-form/apps/main/src/GithubApp/componets/Header/Header.types.ts

And you need to pass form instance in test file for Header component:
https://github.com/Malay-dev/nx-vite-react-ts-mantine-boilerplate/blob/mantine-hook-form/apps/main/src/GithubApp/componets/Header/__tests__/Header.spec.tsx

And rewrite some code in test file

Reply:
@Malay-dev if you need more help text me please 🙏

Reply:
@dipiash Yes.. I am stuck with test cases... I have fixed the debounced value for `repositoryName ` using mantine `useDebouncedValue ` hook... But I am unable to resolve this test file issue...

Reply:
OK, I'll help you in few hours

Reply:
### Test file

For file `apps/main/src/GithubApp/componets/Header/__tests__/Header.spec.tsx` you can do that:
1. render the form hook with default values:
   ```typescript
   const { result: formHookReference } = renderHook(() =>
      useForm<{
        repositoryName: string
        license: string
      }>({
        initialValues: {
          license: '',
          repositoryName: '',
        },
      }),
    )
   ```
2. and use it on `Header` component:
   ```typescript
   <Header form={formHookReference.current} />
   ```
3. and after that you can check the form state
   ```typescript
   expect(formHookReference.current.values.repositoryName).toBe('react')
   ```

**The full example of test file:**
```typescript
import React from 'react'
import { describe, expect, it } from 'vitest'

import { MockedProvider, MockedResponse } from '@apollo/react-testing'
import { useForm } from '@mantine/form'
import { ListLicensesDocument } from '@nx-vite-react-ts-mantine-boilerplate/graphql'
import { ThemeProvider } from '@nx-vite-react-ts-mantine-boilerplate/ui-kit'
import { act, fireEvent, render, renderHook, screen, waitFor } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'

import { Header } from '../index'

import licensesListMockDataSuccess from './mocks/result.success.json'

const mocks: {
  success: readonly MockedResponse[]
  error: readonly MockedResponse[]
} = {
  error: [
    {
      error: new Error('An error occurred'),
      request: {
        query: ListLicensesDocument,
      },
    },
  ],
  success: [
    {
      request: {
        query: ListLicensesDocument,
      },
      result: {
        data: licensesListMockDataSuccess,
      },
    },
  ],
}

describe('Header', () => {
  it('success render', async () => {
    const user = userEvent.setup()

    const { result: formHookReference } = renderHook(() =>
      useForm<{
        repositoryName: string
        license: string
      }>({
        initialValues: {
          license: '',
          repositoryName: '',
        },
      }),
    )

    render(
      <ThemeProvider>
        <MockedProvider mocks={mocks.success}>
          <Header form={formHookReference.current} />
        </MockedProvider>
      </ThemeProvider>,
    )

    await act(() => new Promise((resolve) => setTimeout(resolve, 0))) // wait for response

    const elementLoading = screen.getByTestId('licenses-select-loading')

    expect(elementLoading).toBeInTheDocument()

    // Change value in select element
    const selectElement = screen.getByTestId('licenses-select')

    expect(selectElement).toBeInTheDocument()
    expect(selectElement).toHaveValue('--- Not Selected ---')

    await user.click(selectElement)
    await user.click(screen.getByText('MIT License'))

    expect(formHookReference.current.values.license).toBe('mit')
    expect(selectElement).toHaveValue('MIT License')

    // Change text in input field
    const inputElement = screen.getByTestId('search-by-name')

    expect(inputElement).toBeTruthy()
    expect(inputElement).toHaveValue('')

    fireEvent.change(inputElement, { target: { value: 'react' } })

    await waitFor(() => {
      expect(inputElement).toHaveValue('react')
    })

    expect(formHookReference.current.values.repositoryName).toBe('react')
  })

  it('error render', async () => {
    const { result: formHookReference } = renderHook(() =>
      useForm<{
        repositoryName: string
        license: string
      }>({
        initialValues: {
          license: '',
          repositoryName: '',
        },
      }),
    )

    render(
      <ThemeProvider>
        <MockedProvider mocks={mocks.error}>
          <Header form={formHookReference.current} />
        </MockedProvider>
      </ThemeProvider>,
    )

    await act(() => new Promise((resolve) => setTimeout(resolve, 0))) // wait for response

    screen.getByText('Search by repo name')
    screen.getByText('Licenses loading error.')
  })
})
```

### Declarations file
I see that after this changes test failed with error for css modules declaration.

Please put the file `apps/main/declarations.d.ts` with content:
    ```typescript
    declare module '*.css' {
       const content: Record<string, string>
       export default content
     }
     ```
it's help to TS to clarify css modules files in tests.

Let me know if it clear explanation please